src/main/java/uk/ac/cam/cl/dtg/segue/scheduler/SegueJobService.java

```diff
@@ -180,6 +193,17 @@ public synchronized void initialiseService() throws SchedulerException {
         }
     }
 
+    /**
+     * Attempt to complete running jobs, then shut down the scheduler.
+     * @throws SchedulerException on failure to terminate.
+     */
+    public synchronized void shutdownService() throws SchedulerException {
+        if (wasStarted()) {
+            log.info("Shutting down Segue Job Service");
+            this.scheduler.shutdown(true);
```

Comment:
Cool, and passing in `true` means it will wait for all running jobs to complete first. (Your javadoc does convey this).